### PR TITLE
Renew expired certs for TestCertOptionsAndRetrieveID

### PR DESCRIPTION
**Pull request by Shikugawa**
Signed-off-by: Rei Shimizu <rei@tetrate.io>

ECC certs for `TestCertOptionsAndRetrieveID` are expired then all unit tests are failed. In this PR, it does renewal of them and add `san` option to provide DNS name fields on X509 cert for generating valid one. It also adds missing steps to generate certs for unittest.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[x] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[x] Does not have any changes that may affect Istio users.

### CODE REVIEW

File: security/pkg/pki/testdata/README.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+## ECC root certificate
+
+```
+go run main.go -ec-sig-alg ECDSA -ca true
+```
+
+## ECC client certificate signed by root certificate
+
+```
+go run main.go -ec-sig-alg ECDSA -san watt -signer-cert ../../pkg/pki/testdata/ec-root-cert-new.pem -signer-priv ../../pkg/pki/testdata/ec-root-key-new.pem
```
Comment (howardjohn) [inline security/pkg/pki/testdata/README.md:14]: what is the expiration? can we make it 10yr?

**Review by howardjohn — APPROVED** [review]

File: security/pkg/pki/testdata/README.md (hunk 1 above)
Comment (Shikugawa) [reply security/pkg/pki/testdata/README.md]: Change script default valiue as 10yrs

**Comment by ramaraochavali** [comment]
@Shikugawa Thanks for fixing this and adding that readme doc.

**Pull request merged**